plugins/vlc-video/vlc-video-source.c

```diff
@@ -382,6 +382,57 @@ static void vlcs_video_display(void *data, void *picture)
 	UNUSED_PARAMETER(picture);
 }
 
+static void calculate_display_size(struct vlc_source *c, unsigned *width,
+				   unsigned *height)
+{
+	libvlc_media_t *media = libvlc_media_player_get_media_(c->media_player);
+
+	if (!media)
+		return;
+
+	libvlc_media_track_t **tracks;
+
+	unsigned count = libvlc_media_tracks_get_(media, &tracks);
+
+	if (count > 0) {
+		for (unsigned i = 0; i < count; i++) {
+			libvlc_media_track_t *track = tracks[i];
+
+			if (track->i_type != libvlc_track_video)
+				continue;
+
+			int display_width = track->video->i_width;
+			int display_height = track->video->i_height;
+
+			if (display_width == 0 || display_height == 0)
+				continue;
+
+			/* Adjust for Sample Aspect Ratio (SAR) */
+			if (track->video->i_sar_num > 0 &&
+			    track->video->i_sar_den > 0) {
+				display_width = display_width *
+						track->video->i_sar_num /
+						track->video->i_sar_den;
+			}
+
+			/* ORIENT_IS_SWAP definition taken from VLC */
+			if ((track->video->i_orientation & 4) != 0) {
```

Comment:
Please include the vlc_es header and use `ORIENT_IS_SWAP()` directly.

Reply:
I couldn't figure out how to do that, or even if I should since it doesn't appear to be part of the exported libvlc stuff, hence copying the logic here. I just spent another chunk of time trying to figure it out and failed and I'm going to have to give up at this point. If it's easy/appropriate do to could you point me in the right direction please?

Reply:
I've pushed a different, more explicit implementation that I actually prefer to VLC's own - I never really liked the obscure bitmask check anyway, and this one doesn't require anything that's not included with libvlc.h. Will that do?